rotkehlchen/chain/ethereum/modules/nft/nfts.py

```diff
@@ -172,29 +175,42 @@ def get_all_info(
 
     def get_db_nft_balances(self, filter_query: 'NFTFilterQuery') -> dict[str, Any]:
         """Filters (with `filter_query`) and returns cached nft balances in the nfts table"""
-        query, bindings = filter_query.prepare()
         total_usd_value = ZERO
+        query, bindings = filter_query.prepare(with_pagination=False)
         with self.db.conn.read_ctx() as cursor:
-            cursor.execute(  # this sometimes causes https://github.com/rotki/rotki/issues/5432
-                'SELECT identifier, name, last_price, last_price_asset, manual_price, is_lp, '
-                'image_url, collection_name FROM nfts ' + query,
-                bindings,
-            )
-            entries = [_deserialize_nft_from_db(x) for x in cursor]
-            query, bindings = filter_query.prepare(with_pagination=False)
+            entries_total = cursor.execute('SELECT COUNT(*) FROM nfts').fetchone()[0]
+            entries_found = 0
+            usd_price_tuples = {}  # store the queried prices and calculated usd price
             cursor.execute(
-                'SELECT last_price, last_price_asset FROM nfts ' + query,
+                'SELECT last_price, last_price_asset, identifier FROM nfts ' + query,
                 bindings,
             )
-            entries_found = 0
-            for db_entry in cursor:
-                _, _, usd_price = _deserialize_nft_price(
+            for db_entry in cursor:  # we do this query because the previous one is limited to the page and we need to query the total usd price  # noqa: E501
+                price_in_asset, price_asset, usd_price = _deserialize_nft_price(
                     last_price=db_entry[0],
                     last_price_asset=db_entry[1],
                 )
+                usd_price_tuples[db_entry[2]] = (price_in_asset, price_asset, usd_price)  # save all the prices to reuse them when creating the response  # noqa: E501
                 total_usd_value += usd_price
                 entries_found += 1
-            entries_total = cursor.execute('SELECT COUNT(*) FROM nfts').fetchone()[0]
+
+        with self.db.user_write() as write_ctx:
+            write_ctx.executemany(
+                'UPDATE nfts SET usd_price=? WHERE identifier=?',
+                [(float(price_tuple[2]), ident) for ident, price_tuple in usd_price_tuples.items()],  # noqa: E501
+            )  # update the usd value since the consumer of the api always sorts by usd value
+
+        query, bindings = filter_query.prepare()
+        with self.db.conn.read_ctx() as cursor:
+            cursor.execute(  # this sometimes causes https://github.com/rotki/rotki/issues/5432
```

Comment:
This is from before or did you see it now?

Reply:
This is from before, I just moved the line at the start of the logic since is needed later